src/Http/Http/src/Internal/RequestCookieCollection.cs

```diff
@@ -19,20 +21,21 @@ internal class RequestCookieCollection : IRequestCookieCollection
         private static readonly IEnumerator<KeyValuePair<string, string>> EmptyIEnumeratorType = EmptyEnumerator;
         private static readonly IEnumerator EmptyIEnumerator = EmptyEnumerator;
 
-        private Dictionary<string, string>? Store { get; set; }
+        private AdaptiveCapacityDictionary<string, string> Store { get; set; }
 
         public RequestCookieCollection()
         {
+            Store = new AdaptiveCapacityDictionary<string, string>(StringComparer.OrdinalIgnoreCase);
         }
 
-        public RequestCookieCollection(Dictionary<string, string> store)
+        public RequestCookieCollection(int capacity)
         {
-            Store = store;
+            Store = new AdaptiveCapacityDictionary<string, string>(capacity, StringComparer.OrdinalIgnoreCase);
         }
 
-        public RequestCookieCollection(int capacity)
+        public RequestCookieCollection(Dictionary<string, string> store)
         {
-            Store = new Dictionary<string, string>(capacity, StringComparer.OrdinalIgnoreCase);
+            Store = new AdaptiveCapacityDictionary<string, string>(store.ToList(), capacity: store.Count, StringComparer.OrdinalIgnoreCase);
```

Comment:
AdaptiveCapacityDictionary should allow taking an existing Dictionary to avoid the copy in this scenario.

Reply:
I'm not that concerned about this path, it's only used in tests. Also, if you pass in a dictionary of size 5, you would need to copy it to the list as well.

Reply:
> Also, if you pass in a dictionary of size 5, you would need to copy it to the list as well.

Isn't it only ever using one collection or the other? And once it switches to the dictionary it only switches back if you call Clear?

Reply:
Yes, but if you pass in a dictionary, do you want to to be copied into another dictionary or use the list?

Reply:
If I pass in a dictionary, then use that as the store, don't copy at all.

Reply:
At that point, why just set the store as the dictionary you passed in?

Reply:
I added a comment saying it's only used in tests.

Reply:
Because the RequestCookiesCollection.Store field for example requires a AdaptiveCapacityDictionary.

Reply:
I can fix that.